backend/packages/Upgrade/src/api/controllers/StratificationController.ts

```diff
@@ -0,0 +1,240 @@
+ *             type: string
+ *           group:
+ *             type: object
+ *           workingGroup:
+ *             type: object
+ *       stratificationFactor:
+ *         type: object
+ *         required:
+ *           - stratificationFactorName
+ *         properties:
+ *           stratificationFactorName:
+ *             type: string
+ *       stratificationFactorValue:
+ *         type: string
+ */
+
+/**
+ * @swagger
+ * tags:
+ *   - name: Stratification
+ *     description: CRUD operations related to Stratification
+ */
+@Authorized()
+@JsonController('/stratification')
+export class StratificationController {
+  constructor(public stratificatonService: StratificationService) {}
+
+  /**
+   * @swagger
+   * /stratification:
+   *    get:
+   *       description: Get all stratification
+   *       tags:
+   *         - Stratification
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Get all stratification
+   *            schema:
+   *              type: array
+   *              items:
+   *                $ref: '#/definitions/FactorStrata'
+   *          '401':
+   *            description: Authorization Required Error
+   */
+  @Get()
+  public async getAllStratification(@Req() request: AppRequest): Promise<FactorStrata[]> {
+    return this.stratificatonService.getAllStratification(request.logger);
+  }
+
+  /**
+   * @swagger
+   * /stratification/download/{factor}:
+   *    get:
+   *      description: Get stratificationFactor CSV by factorName
+   *      tags:
+   *        - Stratification
+   *      produces:
+   *        - application/json
+   *      parameters:
+   *        - in: path
+   *          name: factor
+   *          description: Factor name
+   *          required: true
+   *          schema:
+   *            type: string
+   *      responses:
+   *        '200':
+   *          description: Get stratificationFactor detailed CSV by name
+   *        '401':
+   *          description: Authorization Required Error
+   *        '404':
+   *          description: Factor name not found
+   *        '500':
+   *          description: Internal Server Error, FactorName is not valid
+   */
+  @Get('/download/:factor')
+  public async getStratificationByFactor(
+    @Param('factor') factor: string,
+    @Req() request: AppRequest,
+    @Res() res: express.Response
+  ): Promise<any> {
+    if (!factor) {
+      return Promise.reject(new Error(SERVER_ERROR.MISSING_PARAMS + ' : stratification Factor should not be null.'));
+    }
+
+    const data = await this.stratificatonService.getCSVDataByFactor(factor, request.logger);
+
+    // Convert JSON data to CSV
+    const parser = new Parser();
+    const csv = parser.parse(data);
+
+    // return csv file with appropriate headers to request;
+    res.setHeader('Content-Type', 'text/csv; charset=UTF-8');
+    res.setHeader('Content-Disposition', 'attachment; filename="data.csv"');
```

Comment:
Instead of just data.csv you can also add factor name in the csv file name. Something like this -> data-factorName.csv.